src/mixins/nft.js

```diff
@@ -364,6 +364,12 @@ export default {
         await this.initIfNecessary();
         this.fetchUserCollectedCount();
         this.userAccountBalanceFetch = getAccountBalance(this.getAddress);
+        logPurchaseFlowEvent(this, 'add_to_cart', {
+          txHash,
```

Comment:
these is no `txHash` yet.
Also what does `initIfNecessary` do, should we put the log before it?